### PR TITLE
Initial support for afta command

### PR DESCRIPTION
**Pull request by oddcoder**
for doing aftm on all functions

I prefered no to use apis here because aei, aeim have almost no api

### CODE REVIEW

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -39,9 +39,81 @@ static bool anal_is_bad_call(RCore *core, ut64 from, ut64 to, ut64 addr, ut8 *bu
...
+	char uri[32];
+	char nomalloc[256];
+	char *p;
+	if (*input == '?') {
+		eprintf ("Usage: [addr] [size] [name]\n");
+		eprintf ("Default: 0x100000 0xf0000\n");
+		return;
+	}
+
+	p = strncpy (nomalloc, input, 255);
```
Comment (radare) [inline libr/core/cmd_anal.c]: wtf why strncpy? why 256?. try not to use hardcoded limits. strdup() is way better

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -39,9 +39,81 @@ static bool anal_is_bad_call(RCore *core, ut64 from, ut64 to, ut64 addr, ut8 *bu
...
+	if ((p = strchr (p, ' '))) {
+		while (*p == ' ') p++;
+		addr = r_num_math (core->num, p);
+		if ((p = strchr (p, ' '))) {
+			while (*p == ' ') p++;
```
Comment (radare) [inline libr/core/cmd_anal.c]: use r_str_chop() and similars.

File: libr/core/cmd_anal.c [hunk 3]
```diff
@@ -39,9 +39,81 @@ static bool anal_is_bad_call(RCore *core, ut64 from, ut64 to, ut64 addr, ut8 *bu
...
+			size = (ut32)r_num_math (core->num, p);
+			if (size < 1)
```
Comment (radare) [inline libr/core/cmd_anal.c]: missing braces. also if size is ut32 the min value will be 0. what about making that signed to catch >2GB as invalid?

File: libr/core/cmd_anal.c [hunk 4]
```diff
@@ -39,9 +39,81 @@ static bool anal_is_bad_call(RCore *core, ut64 from, ut64 to, ut64 addr, ut8 *bu
...
+				size = 0xf0000;
+			if ((p = strchr (p, ' '))) {
+				while (*p == ' ') p++;
+				snprintf (name, 128, "mem.%s", p);
```
Comment (radare) [inline libr/core/cmd_anal.c]: use r_str_newf. dont snprintf . even sdb_fmt would be better option

File: libr/core/cmd_anal.c [hunk 5]
```diff
@@ -39,9 +39,81 @@ static bool anal_is_bad_call(RCore *core, ut64 from, ut64 to, ut64 addr, ut8 *bu
 }
 #endif
 
+static void cmd_esil_mem(RCore *core, const char *input) {
```
Comment (radare) [inline libr/core/cmd_anal.c]: make it return bool. the function have several return points. try to return false if it fails

File: libr/core/cmd_anal.c [hunk 6]
```diff
@@ -50,13 +122,27 @@ static void type_cmd_help (RCore *core) {
...
 		eprintf ("cant find function here\n");
 		return;
 	}
+	RListIter *it;
+	ut64 seek;
 	switch (*input) {
+	case 'a':
+		seek = core->offset;
+		r_list_foreach (core->anal->fcns, it, fcn) {
+			r_core_cmdf (core, "s 0x%08"PFMT64x, fcn->addr);
```
Comment (radare) [inline libr/core/cmd_anal.c]: use r_core_seek

File: libr/core/cmd_anal.c [hunk 7]
```diff
@@ -50,13 +122,27 @@ static void type_cmd_help (RCore *core) {
...
+			r_core_cmd0 (core, "aei");
```
Comment (radare) [inline libr/core/cmd_anal.c]: aei should be executed ONCE. so put this before the loop

File: libr/core/cmd_anal.c [hunk 8]
```diff
@@ -50,13 +122,27 @@ static void type_cmd_help (RCore *core) {
...
+			r_core_cmd0 (core, "aeim");
```
Comment (radare) [inline libr/core/cmd_anal.c]: same for aeim

File: libr/core/cmd_anal.c [hunk 9]
```diff
@@ -50,13 +122,27 @@ static void type_cmd_help (RCore *core) {
...
+			r_core_cmd0 (core, "aftm");
+			r_core_cmd0 (core, "aeim-");
+			r_core_cmd0 (core, "aei-");
+		}
+		r_core_cmdf (core, "s 0x%08"PFMT64x, seek);
```
Comment (radare) [inline libr/core/cmd_anal.c]: try to use the api instead of calling commands

File: libr/core/cmd_anal.c [hunk 10]
```diff
@@ -50,13 +122,27 @@ static void type_cmd_help (RCore *core) {
...
+		break;
 	case 'm':
-		r_anal_esil_set_pc (core->anal->esil, fcn? fcn->addr: core->offset);
+		r_anal_esil_set_pc (core->anal->esil, fcn? fcn->addr : core->offset);
```
Comment (radare) [inline libr/core/cmd_anal.c]: drop that space

**Comment by oddcoder** [comment]
I updated the branch and kind of removed (returned it to its original place) lots of code that wasn't clean

**Comment by radare** [comment]
Would be nice to see how much faster is now

**Comment by oddcoder** [comment]
It was fast enough of /bin/ls that I started suspecting that it didn't run on all functions

**Comment by XVilka** [comment]
Next command should be 'nafta' or 'caftan'

**Comment by oddcoder** [comment]
what would that be for ?

**Comment by Maijin** [comment]
joke

**Pull request merged**